src/hoi4_world/military/task_force_template.cpp

```diff
@@ -0,0 +1,58 @@
+#include "src/hoi4_world/military/task_force_template.h"
+
+#include "external/fmt/include/fmt/format.h"
+
+bool hoi4::TaskForceTemplate::AllVariantsActive(const std::set<std::string>& active_variants) const
+{
+   for (const auto& ship: ships_)
+   {
+      if (!active_variants.contains(ship.GetEquipment()))
+      {
+         return false;
+      }
+      if (!active_variants.contains(ship.GetLegacyEquipment()))
+      {
+         return false;
+      }
+      if (!active_variants.contains(ship.GetVersion()))
+      {
+         return false;
+      }
+   }
+   return true;
```

Comment:
You could probably use std::all_of for this

Reply:
I can, but unless I'm missing something it doesn't look like an improvement? I would prefer to keep this one as the plain for loop.